pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator_test.go

```diff
@@ -350,6 +367,829 @@ var _ = Describe("VirtualMachine Mutator", func() {
+
+			vm.Spec.Template.Spec.Volumes = append(vm.Spec.Template.Spec.Volumes, v1.Volume{
+				Name: inferVolumeName,
+				VolumeSource: v1.VolumeSource{
+					DataVolume: &v1.DataVolumeSource{
+						Name: dvWithSourceRef.Name,
+					},
+				},
+			})
+
+			vmSpec, _ := getVMSpecMetaFromResponse()
+			Expect(vmSpec.Instancetype).To(Equal(expectedInstancetypeMatcher))
+			Expect(vmSpec.Preference).To(Equal(expectedPreferenceMatcher))
+		},
+			Entry("and PersistentVolumeClaim for InstancetypeMatcher",
+				pvcName, "PersistentVolumeClaim", k8sv1.NamespaceDefault,
+				&v1.InstancetypeMatcher{
+					InferFromVolume: inferVolumeName,
+				},
+				&v1.InstancetypeMatcher{
+					Name: defaultInferedName,
+					Kind: defaultInferedKind,
+				}, nil, nil,
+			),
+			Entry("and PersistentVolumeClaim for PreferenceMatcher",
+				pvcName, "PersistentVolumeClaim", k8sv1.NamespaceDefault,
+				nil, nil,
+				&v1.PreferenceMatcher{
+					InferFromVolume: inferVolumeName,
+				},
+				&v1.PreferenceMatcher{
+					Name: defaultInferedName,
+					Kind: defaultInferedKind,
+				},
+			),
+			Entry(",DataSource and PersistentVolumeClaim for InstancetypeMatcher",
+				dsWithSourcePVCName, "DataSource", k8sv1.NamespaceDefault,
+				&v1.InstancetypeMatcher{
+					InferFromVolume: inferVolumeName,
+				},
+				&v1.InstancetypeMatcher{
+					Name: defaultInferedName,
+					Kind: defaultInferedKind,
+				}, nil, nil,
+			),
+			Entry(",DataSource and PersistentVolumeClaim for PreferenceMatcher",
+				dsWithSourcePVCName, "DataSource", k8sv1.NamespaceDefault,
+				nil, nil,
+				&v1.PreferenceMatcher{
+					InferFromVolume: inferVolumeName,
+				},
+				&v1.PreferenceMatcher{
+					Name: defaultInferedName,
+					Kind: defaultInferedKind,
+				},
+			),
+			Entry("and DataSource with annotations for InstancetypeMatcher",
+				dsWithAnnotionsName, "DataSource", k8sv1.NamespaceDefault,
+				&v1.InstancetypeMatcher{
+					InferFromVolume: inferVolumeName,
+				},
+				&v1.InstancetypeMatcher{
+					Name: defaultInferedName,
+					Kind: defaultInferedKind,
+				}, nil, nil,
+			),
+			Entry("and DataSource with annotations for PreferenceMatcher",
+				dsWithAnnotionsName, "DataSource", k8sv1.NamespaceDefault,
+				nil, nil,
+				&v1.PreferenceMatcher{
+					InferFromVolume: inferVolumeName,
+				},
+				&v1.PreferenceMatcher{
+					Name: defaultInferedName,
+					Kind: defaultInferedKind,
+				},
+			),
+		)
+
+		DescribeTable("should infer defaults from DataVolumeTemplate, DataVolumeSourceRef", func(sourceRefName, sourceRefKind, sourceRefNamespace string, instancetypeMatcher, expectedInstancetypeMatcher *v1.InstancetypeMatcher, preferenceMatcher, expectedPreferenceMatcher *v1.PreferenceMatcher) {
+			vm.Spec.Instancetype = instancetypeMatcher
+			vm.Spec.Preference = preferenceMatcher
+			vm.Spec.DataVolumeTemplates = append(vm.Spec.DataVolumeTemplates, v1.DataVolumeTemplateSpec{
+				ObjectMeta: k8smetav1.ObjectMeta{
+					Name: "dataVolume",
```

Comment:
Make this name a const?

Reply:
As above, just a template name in the context of the test so I don't think we need a const here.